code/modules/surgery/organs/liver.dm

```diff
@@ -85,3 +85,21 @@
 			damage+=100
 		if(2)
 			damage+=50
+
+/obj/item/organ/liver/cybernetic/upgraded/advanced //hey thats pretty good
+	name = "advanced cybernetic liver"
+	desc = "An upgraded version of the advanced cybernetic liver, this time it can heal the body VIA synthesis of various chemicals."
+
+/obj/item/organ/liver/cybernetic/upgraded/advanced/on_life()
+	..()
+	var/mob/living/carbon/C = owner
+
+	if(istype(C))
+		if(!(C.reagents.has_reagent("mannitol", 10)))
+			C.reagents.add_reagent("mannitol", 10)
+		if(!(C.reagents.has_reagent("earthsblood", 10)))
+			C.reagents.add_reagent("earthsblood", 10)
+		if(!(C.reagents.has_reagent("tricordrazine", 10)))
+			C.reagents.add_reagent("tricordrazine", 10)
+		if(!(C.reagents.has_reagent("omnizine", 5)))
+			C.reagents.add_reagent("omnizine", 5)
```

Comment:
Wow, this liver seems deeply overpowered. Anyone want to weigh in on the balance of automatically adding all the advanced healing substances to someone forever?

Reply:
It'll break the game because you'll instantly heal.
Or it would, except it continually adds those reagents, so you'll actually end up ODing on all of those chems and instantly dying.

If that didn't happen you're basically just invincible. No.